src/main/java/com/direwolf20/buildinggadgets/common/tools/LiquidInteractions.java

```diff
@@ -0,0 +1,56 @@
+package com.direwolf20.buildinggadgets.common.tools;
+
+public class LiquidInteractions {
+    public enum GadgetGeneric {
+        IGNORE, REPLACE_TOP;
+
+        public static byte getID(LiquidInteractions.GadgetGeneric mode) {
+            switch (mode) {
+                case IGNORE:
+                    return (byte) 0;
+                case REPLACE_TOP:
+                    return (byte) 1;
+                default:
+                    return (byte) 0;
+            }
+        }
+        public static LiquidInteractions.GadgetGeneric idToEnum(byte id) {
```

Comment:
Then find it values here (if you want to, cache values, but I don't think that's really important)